plugins/outputs/iotdb/iotdb.go

```diff
@@ -229,6 +231,60 @@ func (s *IoTDB) convertMetricsToRecordsWithTags(metrics []telegraf.Metric) (*rec
 	return rwt, nil
 }
 
+// takes in an array of compiled regex patterns
+// and validates them on the given tag.
+func matchToEncloseInBacktick(regexStrings []*regexp.Regexp, tag string) string {
+	// loops through all the regex patterns and if one
+	// pattern matches returns the tag between `
+	for _, regex := range regexStrings {
+		if regex.MatchString(tag) {
+			return "`" + tag + "`"
+		}
+	}
+
+	return tag
+}
+
+// checks is the tag contains any IoTDB invalid character
+func (s *IoTDB) validateTag(tag string) (string, error) {
+	// matches any word that has a non valid backtick
+	// `word`  							 <- dosen't match
+	// ``word , `wo`rd` , `word , word`  <- match
+	forbiddenBacktick, _ := regexp.Compile("^[^\x60].*?[\x60]+.*?[^\x60]$|^[\x60].*[\x60]+.*[\x60]$|^[\x60]+.*[^\x60]$|^[^\x60].*[\x60]+$")
+	allowedBacktick, _ := regexp.Compile("^[\x60].*[\x60]$")
```

Comment:
If we know this should compile, you should do
```suggestion
	forbiddenBacktick := regexp.MustCompile("^[^\x60].*?[\x60]+.*?[^\x60]$|^[\x60].*[\x60]+.*[\x60]$|^[\x60]+.*[^\x60]$|^[^\x60].*[\x60]+$")
	allowedBacktick := regexp.MustCompile("^[\x60].*[\x60]$")
```
to catch programming errors. We might also want define those two as package-global variables to not compile them for every metric again as they seem constant...